server/py/services/alerts/crud/alerts.py

```diff
@@ -175,76 +175,36 @@ def process_event(
-
-                if alert.criteria.period is not None:
-                    # adjust the sliding window of events
-                    # in case the EventEntityKind is JOB then we should consider the runs monitoring interval here
-                    # because the monitoring runs might miss events occurring just before the interval.
-                    offset = 0
-                    if (
-                        alert.entities.kind
-                        == mlrun.common.schemas.alert.EventEntityKind.JOB
-                    ):
-                        offset = int(mlconfig.monitoring.runs.interval)
-                    self._normalize_events(
-                        state_obj,
-                        framework.utils.helpers.string_to_timedelta(
-                            alert.criteria.period, offset, raise_on_error=False
-                        ),
-                    )
+        if not self._event_entity_matches(alert.entities, event_data.entity):
+            return
 
-                if len(state_obj["events"]) >= alert.criteria.count:
-                    send_notification = True
-            else:
-                send_notification = True
-
-            active = False
-            update_state = True
-            if send_notification:
-                state["count"] += 1
-                logger.debug("Sending notifications for alert", name=alert.name)
-                AlertNotificationPusher().push(alert, event_data)
-
-                if alert.reset_policy == "auto":
-                    self.reset_alert(session, alert.project, alert.name)
-                    services.api.crud.AlertActivation().store_alert_activation(
-                        session, alert, event_data
-                    )
-                    update_state = False
-                else:
-                    active = True
-                    state["active"] = True
-                    self._get_alert_state_cached().cache_replace(
-                        state, session, alert.id
-                    )
+        state_obj = self._states.get(alert.id, {"events": []})
 
-                # we store the state along with the events that triggered the alert
-                framework.utils.singletons.db.get_db().store_alert_state(
-                    session,
-                    alert.project,
-                    alert.name,
-                    count=state["count"],
-                    last_updated=event_data.timestamp,
-                    obj=state_obj,
-                    active=active,
-                )
+        # Append the event timestamp if the alert is active or has criteria.
+        # If the alert is active, it follows a manual reset policy, so we need to continue counting events.
+        # If the criteria is None and the alert is inactive, we trigger the alert immediately and do not cache events.
+        if state["active"] or alert.criteria is not None:
+            state_obj["events"].append(event_data.timestamp)
```

Comment:
```suggestion
            state_obj["events_timestamps"].append(event_data.timestamp)
```
this will make the state_obj more understandable

Reply:
agree!